### PR TITLE
Update BBEs

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
This PR updates the following BBEs
- optional type 
- optional field access, renamed from nil lifting 
- range expression, renamed from integer ranges
- string template
- type cast

Partial fixes for #17065, #17066 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: examples/string-template/string_template.bal [hunk 1]
```diff
@@ -2,8 +2,8 @@ import ballerina/io;
 
 public function main() {
     string name = "Ballerina";
-    // This creates a `string` template.
+    // Create a `string` template, embedding the `name` variable.
```
Comment (praneesha) [inline examples/string-template/string_template.bal]:
```suggestion
    // Create a `string` template embedding the `name` variable.
```

**Comment by hasithaa** [comment]
@MaryamZi There are few open review comments, Please fix them in a separate PR.

File: examples/optional-field-access/optional_field_access.bal [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    // Now `age` will be the `int` value set.
+    age = p1?.age;
+    io:println("Age: ", age);
+
+    Person? p2 = getPerson("John", 30);
+    io:println(p2);
+
+    // The optional field access operator is a lifted operator in Ballerina.
+    // The types of both `p2` and `p2.addr` are optional record types.
+    // If the type of either `p2` or `p2.addr` is `()` during the run time, `()` would
```
[the remark this answers is not in the document]
Comment (MaryamZi) [reply examples/optional-field-access/optional_field_access.bal]: What we are trying to check is if either `p2` or `p2.addr` is `()`.

[the remark this answers is not in the document]
Comment (MaryamZi) [reply examples/optional-field-access/optional_field_access.bal]: Will change to 
```suggestion
    // If the type of either `p2` or `p2.addr` is `()` at run time, `()` would
```

[the remark this answers is not in the document]
Comment (praneesha) [reply examples/optional-field-access/optional_field_access.bal]: +1

**Pull request merged**